llama_hub/agent_search/base.py

```diff
@@ -0,0 +1,75 @@
+"""AgentSearch reader
+
+Example as of 1/8/2024:
+
+```python
+AgentSearch = download_loader("AgentSearch")
+
+document = reader.load_data(
+    query="latest news",
+    search_provider="bing"
+)[0]
+
+print(f'Document:\n{document} ')
+```
+
+```plaintext
+Document:
+Doc ID: 67a57dfe-8bd6-4c69-af9d-683e76177119
+Text: The latest news encompasses a wide array of topics, reflecting
+the dynamic and complex nature of the world today. Notable events
+include the conviction of a man for killing his ex-wife's new partner,
+highlighting the ongoing issue of domestic violence and its legal
+consequences [2]. In the realm of international relations, the release
+of Jeffrey... 
+```
+
+For more information, refer to the docs:
+https://agent-search.readthedocs.io/en/latest/
+"""
+
+from typing import Any, Dict, List, Optional
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+
+
+class AgentSearchReader(BaseReader):
+    """AgentSearch reader."""
+
+    def __init__(
+        self,
+        api_base: Optional[str] = None,
+        api_key: Optional[str] = None,
+    ):
+        """Initialize with parameters."""
+        import_err_msg = (
+            "`agent-search` package not found, please run `pip install agent-search`"
+        )
+        try:
+            import agent_search  # noqa: F401
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        from agent_search import SciPhi
+
+        self._client = SciPhi(api_base=api_base, api_key=api_key)
```

Comment:
Can the API key also be an env var? Does the api_base default to something internally?

I noticed in the README that the API key and API base was not mentioned

Reply:
SciPhi defaults to SCIPHI_API_KEY if no api_key is provided, this just gives the user additional optionality